block_pickup.php

```diff
@@ -160,29 +134,16 @@ public function fetch_recent_courses() : array {
             $course = new stdClass();
             $course->fullname = $cr->fullname;
             $course->viewurl = new moodle_url('/course/view.php', array('id' => $cr->id));
-            $course->coursecategory = $cr->name;
+            $course->coursecategory = core_course_category::get($cr->category)->get_formatted_name();
 
             /* Progress. */
-            if ($cr->enablecompletion) {
-                $percentage = \core_completion\progress::get_course_progress_percentage($cr, $USER->id);
-                if (!is_null($percentage)) {
-                    $percentage = floor($percentage);
-                    $course->progress = $percentage;
-
-                }
+            if ($percentage = progress::get_course_progress_percentage($cr, $USER->id)) {
```

Comment:
`enablecompletion ` is checked inside the `\core_completion\progress::get_course_progress_percentage` function, so no need to check it before.